packages/client/src/components/ReviewsDisplay/ReviewsDisplay.jsx

```diff
@@ -1,51 +1,68 @@
 import React, { useEffect, useState } from 'react';
-import {
-  Box,
-  List,
-  ListItem,
-  Divider,
-  ListItemAvatar,
-  Avatar,
-  Typography,
-  Button,
-  CircularProgress,
-} from '@mui/material';
+import { Box, List, Divider, Button, CircularProgress } from '@mui/material';
 
-import AddCircleIcon from '@mui/icons-material/AddCircle';
-import ThumbDownOffAltIcon from '@mui/icons-material/ThumbDownOffAlt';
-import ThumbUpIcon from '@mui/icons-material/ThumbUp';
-import { RatingStars } from '../RatingStars/RatingStars';
 import ArrowDropDownIcon from '@mui/icons-material/ArrowDropDown';
 import ArrowDropUpIcon from '@mui/icons-material/ArrowDropUp';
 import { MovieDetailsLayout } from '../../containers/MovieDetailsLayout/MovieDetailsLayout';
 import styled from '@emotion/styled';
 import { apiURL } from '../../apiURL';
+import { ReviewItem } from './ReviewItem';
+import { HeaderReview } from './HeaderReview';
+import { useUserContext } from '../../context/UserContext';
 
 export function ReviewsDisplay({ movieID }) {
+  const { user } = useUserContext();
   const [reviews, setReviews] = useState([]);
   const [isLoading, setIsLoading] = useState(false);
   const [showAllReviews, setShowAllReviews] = useState(false);
+  const [userReview, setUserReview] = useState(null);
 
   useEffect(() => {
     if (!movieID) return;
     const fetchReviewsList = async () => {
       setIsLoading(true);
-      if (movieID) {
-        try {
-          const response = await fetch(`${apiURL()}/reviews/movie/${movieID}`);
-          const data = await response.json();
-          setReviews(data);
-        } catch (error) {
-          setIsLoading(false);
-        }
+      try {
+        const response = await fetch(`${apiURL()}/reviews/movie/${movieID}`);
+        const data = await response.json();
+        setReviews(data);
+      } catch (error) {
+        setIsLoading(false);
       }
       setIsLoading(false);
     };
     fetchReviewsList();
   }, [movieID]);
+
+  useEffect(() => {
+    if (!movieID || !user) return;
+    const fetchUserReview = async () => {
+      try {
+        const userReviewResponse = await fetch(
+          `${apiURL()}/reviews/${movieID}/uid/${user.uid}`,
```

Comment:
@Kositthai in regard to my comments to your PR, here is something I don't understand again about multiple IDs (I doesn't mean it's wrong, I just don't understand that  :)).

Here is an endpoint caching review by uuid and there I saw endpoint catching favorites by userID. Is this because IDs are not streamlined or because I don't get something 🤯 ? :D

Reply:
if i use user.id its returning empty array even though there is reviews for this user, and it return it if i user user,uid

Reply:
The different ids come because of the implementation  of firebase authentication.
Landing team put user object inside the app's context and that user has only firebase native uid field. Then they put this uid in users table and it creates users.id field. 
All the other tables have this users.id as a foreign key. So we need to fetch everything by first mapping uid to users.id and then we can fetch id to work with it

Reply:
ok, thanks! Now I get it :)